src/gravitational_waves/gw.cpp

```diff
@@ -51,98 +78,184 @@ GravitationalWave::~GravitationalWave()
 {
 }
 
+double GetHtauSH(const double HR, const double K_sw)
+{
+  return 2. / std::sqrt(3) * HR / std::sqrt(K_sw);
+}
+
+double GetHtauSW(const double HR, const double K_sw)
+{
+  return min(GetHtauSH(HR, K_sw), 1.);
+}
+
+double GetYpsilon(const double HR, const double K_sw)
+{
+  return 1. - 1. / sqrt(1. + 2. * GetHtauSW(HR, K_sw));
+}
+
 double GravitationalWave::CalcEpsTurb(double epsturb_in)
 {
   if (epsturb_in == -1)
   {
-    double HtauSW = 2. / std::sqrt(3) * data.HR / std::sqrt(data.K_sw);
-    return std::pow((1 - std::min(HtauSW, 1.)), 2. / 3.);
+    return std::pow((1 - GetYpsilon(data.HR, data.K_sw)), 2. / 3.);
   }
   else
   {
     return epsturb_in;
   }
 }
 
-void GravitationalWave::CalcPeakFrequencySoundWave()
+void GravitationalWave::CalcPeakCollision()
 {
-  double res = 26e-6 * (1. / this->data.HR) *
-               (this->data.transitionTemp / 100) *
-               std::pow(this->data.gstar / 100, 1. / 6);
-  this->data.fPeakSoundWave = res;
+  const double n1 = 2.4;
+  const double n2 = -2.4;
+  const double a1 = 1.2;
+
+  data.CollisionParameter.n1 = n1;
+  data.CollisionParameter.n2 = n2;
+  data.CollisionParameter.a1 = a1;
+
+  // Calculate characteristic frequency
+
+  const double f_p            = 0.11 * data.Hstar0 * data.betaH;
+  data.CollisionParameter.f_b = f_p * pow(-n1 / n2, -1 / a1);
+
+  // Calculate amplitude for collisions
+  const double A_str  = 0.05;
+  const double Ktilde = data.kappa_col * GetKtilde(data.PTStrength);
+  const double Omega_p =
+      data.FGW0 * A_str * pow(Ktilde, 2) / pow(data.betaH, 2);
+
+  data.CollisionParameter.Omega_b =
+      Omega_p * pow(1. / 2. * pow(-n2 / n1, n1 / (n1 - n2)) +
+                        1. / 2. * pow(-n1 / n2, -n2 / (n1 - n2)),
+                    (n1 - n2) / a1);
 }
 
-void GravitationalWave::CalcPeakAmplitudeSoundWave()
+double GravitationalWave::CalculateXiShell()
 {
-  double ratio = 0;
-  if (IsFluidTurnoverApproxOne(this->data.HR, this->data.K_sw))
-  {
-    ratio = this->data.HR * std::pow(this->data.K_sw, 2.);
-  }
+  double XiFront, XiRear;
+  if (data.vw > data.vCJ)
```

Comment:
```suggestion
  if (data.vw >= data.vCJ)
```
.. to match Eq. (3.73) in the paper.